apps/src/lib/node/ledger/ethereum_node/test_tools/mod.rs

```diff
@@ -206,6 +215,13 @@ pub mod mock_web3_client {
                         client.events.push((event_ty, data, height, seen));
                     }
                 }
+                if client.last_block_processed < Some(block_to_check.clone()) {
```

Comment:
```suggestion
                if client.last_block_processed.as_ref() < Some(&block_to_check) {
```
I think we might even be able to compare by ref here

Reply:
yeah, jacob was also surprised about this lol. it's definitely a bit of an obscure-ish feature of `Option` types, I think.